src/languageservice/services/yamlCompletion.ts

```diff
@@ -1113,9 +1113,9 @@ export class YamlCompletion {
           case 'number':
           case 'integer':
           case 'anyOf': {
-            let value = propertySchema.default || propertySchema.const;
-            if (value) {
-              if (type === 'string') {
+            let value = propertySchema.default === undefined ? propertySchema.const : propertySchema.default;
```

Comment:
fixes condition to allow null,0,emptyString as a default or const
the same logic is correctly used in other places